test/gtest/ucp/test_ucp_am.cc

```diff
@@ -1572,8 +1565,8 @@ class test_ucp_am_nbx_rndv : public test_ucp_am_nbx_prereg {
     {
         test_ucp_am_nbx_rndv *self = reinterpret_cast<test_ucp_am_nbx_rndv*>(arg);
 
-        EXPECT_FALSE(self->m_am_received);
-        self->m_am_received = true;
+        EXPECT_LT(self->m_recv_counter, self->m_send_counter);
```

Comment:
EXPECT_EQ(0, m_recv_counter)

Reply:
I can change it because it's non common code. Will be more clear, although the conditions has the same logic